### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by ThinkTankShark**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  return result;
+}
+
+/**
+ * Creates a duplicate-free version of an array
+ * @param     {Array}         val
+ * @return    {Immutable.Set}
+ */
+export function unique(val) {
+  return new Immutable.Set(val);
```
Comment (isuttell) [inline src/utils.js]: I don't want to rely on immutable here. I think it should be a separated

File: tasks/config/webpack.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * ---------------------------------------------------------------
+ *
+ * Module loader, JSX transform, Minification, Sourcemaps
+ *
+ * For usage docs see:
+ *    https://github.com/webpack/grunt-webpack
+ */
+
+var assign = require('object-assign');
```
Comment (isuttell) [inline tasks/config/webpack.js]: Don't need this module any more. Let's remove it.

File: tasks/config/webpack.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      'react': 'React'
+    }
+  });
+
+  // Remove Extract Plugin. Gotta clone to prevent changing above config
+  serverOptions.module = assign({}, serverOptions.module);
+  serverOptions.module.loaders = serverOptions.module.loaders.slice(0);
+  serverOptions.module.loaders.splice(serverOptions.module.loaders.length - 1);
+
+  grunt.config.set('webpack-dev-server', {
```
Comment (isuttell) [inline tasks/config/webpack.js]: Are we even using the dev-server? If not, let's remove it

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends" : "ship-components",
+   "rules": {
+      "indent": 0
```
Comment (ThinkTankShark) [inline .eslintrc]: Indentation seems to be setup differently in different text editors.... I've noticed my tab space seems to be counted as 8 spaces and not a tab.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  "keywords": [
+    "javascript",
+    "javascript-library",
+    "node",
+    "grunt",
+    "webpack",
+    "utilities",
+    "jest",
+    "jasmine"
```
Comment (ThinkTankShark) [inline package.json]: Removed

File: .babelrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (ThinkTankShark) [inline .babelrc]: Switched

File: src/__tests__/strings-spec.js [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+      expect(result).toEqual(str2);
+    });
+  });
+
+  describe('String format Methods', () => {
+    const {capitalize, titleCase, camelCase, toUnderscoreCase} = require('../strings');
+
+    test('should return true when capitalize method converts correctly', () => {
+      let str1 = 'test test';
+      let str2 = 'Test Test';
```
Comment (ThinkTankShark) [inline src/__tests__/strings-spec.js]: The capitalize function will convert the string to a Title Case if more than one string's provided. So One word = Capitalize, 2+ words: Title Case

File: src/__tests__/strings-spec.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+      let str2 = 'thisIsARandomString';
+      let result = camelCase(str1);
+
+      expect(result).not.toEqual(str1);
+      expect(result).toEqual(str2);
+    });
+
+    test('should return true when toUnderscoreCase method converts correctly', () => {
+      let str1 = 'this is a random string';
+      let str2 = 'THIS IS A RANDOM STRING';
```
Comment (ThinkTankShark) [inline src/__tests__/strings-spec.js]: Good catch...Fixed the func and the tests

File: src/__tests__/sort-spec.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        name: 'sample3',
+        desc: 'disc3',
+        updated_at: Date.now() - 50
+      }, {
+        name: 'sample4',
+        desc: 'disc4',
+        updated_at: Date.now() + 20
+      }];
+
+      let arr2 = Object.assign({}, arr1);
```
Comment (ThinkTankShark) [inline src/__tests__/sort-spec.js]: Correct, fixed.

File: src/__tests__/sort-spec.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        age: 30
+      }, {
+        name: 'sample3',
+        age: 80
+      }, {
+        name: 'sample4',
+        age: 20
+      }];
+
+      let arr2 = Object.assign({}, arr1);
```
Comment (ThinkTankShark) [inline src/__tests__/sort-spec.js]: Fixed

File: src/strings.js [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+/**
+ * Validate strings (name, username or email)
+ *
+ * @param     {String}    prop [default=email]
+ * @param     {String}    str
+ * @return    {Bool}
+ */
+export function stringIsValid(prop, str) {
```
Comment (ThinkTankShark) [inline src/strings.js]: This is basically a general template to validate usernames, emails and name for forms....

File: src/utils.js (hunk 1 above)
Comment (ThinkTankShark) [inline src/utils.js]: Removed

File: tasks/config/webpack.js (hunk 2 above)
Comment (ThinkTankShark) [inline tasks/config/webpack.js]: Removed

File: tasks/config/webpack.js (hunk 3 above)
Comment (ThinkTankShark) [inline tasks/config/webpack.js]: Originally was thinking about adding a example folder, but then decided to go with a complete readme file instead, so forgot to remove it. Removed now.

File: src/__tests__/strings-spec.js [hunk 12]
```diff
@@ -0,0 +1,162 @@
+/* eslint-disable */
+
+describe('Strings Library', () => {
+  describe('hash Method', () => {
+    const {hash} = require('../strings');
+
+    test('should return true if 32bit integer generated', () => {
```
Comment (ThinkTankShark) [inline src/__tests__/strings-spec.js]: Not quite sure what it means...

Comment (ThinkTankShark) [inline src/__tests__/strings-spec.js]: Added

File: .eslintrc (hunk 4 above)
Comment (isuttell) [inline .eslintrc]: I think that's why I went with spaces

File: src/__tests__/strings-spec.js (hunk 7 above)
Comment (isuttell) [inline src/__tests__/strings-spec.js]: Then that's a bug. If you want Title case use the title case function. Capitalize should only capitalize the first letter.

Comment (ThinkTankShark) [inline src/__tests__/strings-spec.js]: Fixed

**Pull request merged**